doc/source/roadmap.rst

```diff
@@ -0,0 +1,115 @@
+Each array stores Python strings. While pragmatic, since we rely on NumPy
+for storage and Python for string operations, this is memory inefficient
+and slow. We'd like to provide a native string type for pandas.
+
+The most obvious alternative is Apache Arrow. Currently, Arrow provides
+storage for string data. We can work with the Arrow developers to implement
+the operations needed for pandas users (for example, ``Series.str.upper``).
+
+Apache Arrow Interoperability
+-----------------------------
+
+`Apache Arrow <https://arrow.apache.org>`__ is a cross-language development
+platform for in-memory data. The Arrow logical types are closely aligned with
+typical pandas use cases (for example, support for nullable integers).
+
+We'd like have a pandas DataFrame be backed by Arrow memory and data types
+by default. This should simplify pandas internals and ensure more consistent
+handling of data types through operations.
+
+Block Manager Rewrite
+---------------------
+
+We'd like to replace pandas current internal data structures (a collection of
+1 or 2-D arrays) with a simpler collection of 1-D arrays.
+
+Pandas internal data model is quite complex. A DataFrame is made up of
+one or more 2-dimension "blocks", with one or more blocks per dtype. This
+collection of 2-D arrays is managed by the BlockManager.
+
+The primary benefit of the BlockManager is improved performance on certain
+operations (construction from a 2D array, binary operations, reductions across the columns),
+especially for wide DataFrames. However, the BlockManager substantially increases the
+complexity and maintenance burden of pandas'.
+
+By replacing the BlockManager we hope to achieve
+
+* Substantially simpler code
+* Easier extensibility with new logical types
+* Better user control over memory use and layout
+* Improved microperformance
+* Option to provide a C / Cython API to pandas' internals
+
+See `these design documents <https://dev.pandas.io/pandas2/internal-architecture.html#removal-of-blockmanager-new-dataframe-internals>`__
+for more.
+
+Weighted Operations
+-------------------
+
+In many fields, sample weights are necessary to correctly estimate population
+statistics. We'd like to support weighted operations (like `mean`, `sum`, `std`,
+etc.), possibly with an API similar to `DataFrame.groupby`.
+
+See https://github.com/pandas-dev/pandas/issues/10030 for more.
+
+Package Docstring Validation
+----------------------------
+
+To improve the quality and consistency of pandas docstrings, we've developed
+tooling to check docstrings in a variety of ways.
+https://github.com/pandas-dev/pandas/blob/master/scripts/validate_docstrings.py
+contains the checks.
+
+Like many other projects, pandas uses the
+[numpydoc](https://numpydoc.readthedocs.io/en/latest/) style for writing
+docstrings. With the collaboration of the numpydoc maintainers, we'd like to
+move the checks to a package other than pandas so that other projects can easily
+use them as well.
```

Comment:
Not sure if technically speaking we can consider moving the validation from pandas to numpydoc part of the pandas roadmap. I'd mention it, probably we'll do part or most of the work, but may be it's worth to have it as a point of a section for all the improvements we'd like to see in the documentation:

- Better structure (almost there, but still some work)
- Good looking and responsive theme
- Unify the website and the docs
- Improve the getting started documentation (already started, but I guess makes sense to have it in the roadmap)
- Improve the contributing to pandas documentation
- Standardize format of all docstrings
- Improve content of docstrings
- Create and define a set of standard datasets to be used in all the examples of the documentation
- Add a blog / planet
- Creating generic package for the validation, and move it to numpydoc (so far just one person shown interest)

Reply:
Added a few documentation items. I think having the "move docstring validation out of pandas" can stand on its own though.